src/main/java/seedu/duke/commands/DeleteCommand.java

```diff
@@ -1,60 +1,144 @@
+    protected static final String DELETE_ALL_FLAG = "all";
 
     // input from user
-    public String taskToDelete;
-    public static ArrayList<Item> combinedItemList = new ArrayList<>();
+    private String itemFlag;
+    private int indexToDelete;
+
+    private boolean isDeleteAll;
+    private boolean isCorrectFormat;
+
+    // create logger
+    private static final Logger logger = Logger.getLogger("Logger");
 
-    // v1.0: deleteCommand deletes purely based on index, i.e. delete [TASK_INDEX]
-    // converted command array to string for future uses, where input may have extra spaces
-    // such as 'delete foo bar'
-    public DeleteCommand(String[] itemDescription) {
-        taskToDelete = Arrays.toString(itemDescription);
+
+    // v2.0: deleteCommand deletes purely based on index, i.e. delete -t/-e [TASK_INDEX]
+    public DeleteCommand(String[] command) {
+
+        isCorrectFormat = true;
+        isDeleteAll = false;
+        logger.setLevel(Level.INFO);
+
+        try {
+            logger.log(Level.INFO, "going to start processing");
+            if (command.length == 1) {
+                throw new DukeException("Please specify what you wish to delete.");
+            }
+            itemFlag = command[1].trim();
+            if (isDeleteAllFlag(itemFlag)) {
+                isDeleteAll = true;
+            } else if (isEventFlag(itemFlag) || isTaskFlag(itemFlag)) {
+                prepareInputs(command);
+            } else {
+                logger.log(Level.WARNING, "processing error");
+                throw new DukeException("Invalid item flag entered. Please specify event '-e' "
+                        + "or task '-t'.");
+            }
+        } catch (DukeException e) {
+            System.out.println(e.getMessage());
+            isCorrectFormat = false;
+        } catch (NumberFormatException e) {
+            System.out.println("PLease enter a number for the item index!");
+            isCorrectFormat = false;
+        }
     }
 
     public CommandResult execute() {
+        if (isCorrectFormat) {
+            if (isDeleteAll) {
+                if (confirmDeleteAll()) {
+                    deleteAllEventsAndTasks();
+                    return new CommandResult("I have deleted everything!");
+                } else {
+                    return new CommandResult("I will not delete anything!");
+                }
+            }
 
-        fillCombinedItemList();
+            String deletedItem;
+            if (isEventFlag(itemFlag)) {
+                assert indexToDelete < eventCatalog.size();
+                deletedItem = deleteEvent(indexToDelete);
+                return new CommandResult("I have deleted this event: " + deletedItem);
+            } else if (isTaskFlag(itemFlag)) {
+                assert indexToDelete < eventCatalog.size();
+                deletedItem = deleteTask(indexToDelete);
+                return new CommandResult("I have deleted this task: " + deletedItem);
```

Comment:
Since the current main list only has events, this shouldn't be able to work as intended.